tests/test_extensions/test_ext_autodoc_type_alias_nitpicky.py

```diff
@@ -0,0 +1,80 @@
+"""Test type alias cross-reference resolution in nitpicky mode."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pytest
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from sphinx.testing.util import SphinxTestApp
+
+
+@pytest.mark.sphinx('html', testroot='ext-autodoc-type-alias-xref')
+def test_type_alias_xref_resolution(app: SphinxTestApp, warning: Any) -> None:
+    """Test that type aliases in function signatures can be cross-referenced.
+
+    This tests the fix for issue #10785 where type aliases documented as py:data
```

Comment:
Can we use the full issue URL here please
```suggestion
    This tests the fix for issue #10785 where type aliases documented as :py:data:
```